pkg/model/components/kubecontrollermanager.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi/loader"
+	metav1 "k8s.io/kubernetes/pkg/apis/meta/v1"
+)
+
+const (
+	defaultAttachDetachReconcileSyncPeriod = time.Minute
+)
+
+// KubeControllerManagerOptionsBuilder adds options for the k-c-m to the model
+type KubeControllerManagerOptionsBuilder struct {
+	Context *OptionsContext
+}
+
+var _ loader.OptionsBuilder = &KubeControllerManagerOptionsBuilder{}
+
+// BuildOptions tests for options to be added to the model
+func (b *KubeControllerManagerOptionsBuilder) BuildOptions(o interface{}) error {
+
+	options := o.(*kops.ClusterSpec)
+
+	if options.KubeControllerManager == nil {
+		options.KubeControllerManager = &kops.KubeControllerManagerConfig{}
+	}
+
+	k8sv148, err := kops.ParseKubernetesVersion("v1.4.8")
+
```

Comment:
Also, it might be easier to call `semver.Parse` or `semver.ParseTolerant` directly